src/base/utils/fs.cpp

```diff
@@ -398,3 +398,47 @@ bool Utils::Fs::isNetworkFileSystem(const QString &path)
 #endif
 }
 #endif // Q_OS_HAIKU
+
+QString Utils::Fs::rootFolder(const QStringList &filePaths)
+{
+    QString rootFolder;
+    for (const QString &filePath : filePaths)
+    {
+        const auto filePathElements = QStringView(filePath).split(u'/');
+        // if at least one file has no root folder, no common root folder exists
+        if (filePathElements.count() <= 1)
+            return {};
+
+        if (rootFolder.isEmpty())
+            rootFolder = filePathElements.at(0).toString();
+        else if (rootFolder != filePathElements.at(0))
+            return {};
+    }
+
+    return rootFolder;
+}
+
+QStringList Utils::Fs::stripRootFolder(const QStringList &filePaths)
```

Comment:
> maybe it is a good idea to mutate the input directly?

But then the external code won't match your "everything should be a constant", will it?

Reply:
>But then the external code won't match your "everything should be a constant", will it?

Yes, but somehow I feel like such operations is like operating/mutating on a data (like `QString::`) and it is not so meaningful to keep the original data. Maybe it is due to the function name.

Inspiration taken from `void QString::chop(int n)` and `QString QString::chopped(int len)`:
It would make sense if you used such names:
`QStringList Utils::Fs::strippedRootFolder(const QStringList &filePaths)`
`QStringList Utils::Fs::addedRootFolder(const QStringList &filePaths, const QString &rootFolder)`

Reply:
> somehow I feel like such operations is like operating/mutating on a data (like `QString::`) and it is not so meaningful to keep the original data.

Switching to such a way since I feel it better.